acceptance/openstack/containerinfra/v1/nodegroups_test.go

```diff
@@ -0,0 +1,181 @@
+
+	clusterID, err := createKubernetesCluster(t, client, clusterTemplate.UUID)
+	th.AssertNoErr(t, err)
+	defer DeleteCluster(t, client, clusterID)
+
+	t.Run("list", func(t *testing.T) { testNodeGroupsList(t, client, clusterID) })
+	t.Run("listone-get", func(t *testing.T) { testNodeGroupGet(t, client, clusterID) })
+}
+
+func testNodeGroupsList(t *testing.T, client *gophercloud.ServiceClient, clusterID string) {
+	allPages, err := nodegroups.List(client, clusterID, nil).AllPages()
+	th.AssertNoErr(t, err)
+
+	allNodeGroups, err := nodegroups.ExtractNodeGroups(allPages)
+	th.AssertNoErr(t, err)
+
+	// By default two node groups should be created
+	th.AssertEquals(t, 2, len(allNodeGroups))
+}
+
+func testNodeGroupGet(t *testing.T, client *gophercloud.ServiceClient, clusterID string) {
+	listOpts := nodegroups.ListOpts{
+		Role: "worker",
+	}
+	allPages, err := nodegroups.List(client, clusterID, listOpts).AllPages()
+	th.AssertNoErr(t, err)
+
+	allNodeGroups, err := nodegroups.ExtractNodeGroups(allPages)
+	th.AssertNoErr(t, err)
+
+	// Should be one worker node group
+	th.AssertEquals(t, 1, len(allNodeGroups))
+
+	ngID := allNodeGroups[0].UUID
+
+	ng, err := nodegroups.Get(client, clusterID, ngID).Extract()
+	th.AssertNoErr(t, err)
+
+	// Should have got the same node group as from the list
+	th.AssertEquals(t, ngID, ng.UUID)
+	th.AssertEquals(t, "worker", ng.Role)
+}
+
+func createKubernetesClusterTemplate(t *testing.T, client *gophercloud.ServiceClient) (*clustertemplates.ClusterTemplate, error) {
+	choices, err := clients.AcceptanceTestChoicesFromEnv()
+	if err != nil {
+		return nil, err
+	}
+
+	name := tools.RandomString("TESTACC-", 8)
+	t.Logf("Attempting to create cluster template: %s", name)
+
+	boolFalse := false
+	createOpts := clustertemplates.CreateOpts{
+		COE:                 "kubernetes",
+		DNSNameServer:       "8.8.8.8",
+		DockerStorageDriver: "overlay2",
+		ExternalNetworkID:   choices.ExternalNetworkID,
+		FlavorID:            choices.FlavorID,
+		FloatingIPEnabled:   &boolFalse,
+		ImageID:             choices.MagnumImageID,
+		MasterFlavorID:      choices.FlavorID,
+		MasterLBEnabled:     &boolFalse,
+		Name:                name,
+		Public:              &boolFalse,
+		RegistryEnabled:     &boolFalse,
+		ServerType:          "vm",
+	}
+
+	res := clustertemplates.Create(client, createOpts)
+	clusterTemplate, err := res.Extract()
+	if err != nil {
+		return nil, err
+	}
+
+	t.Logf("Successfully created cluster template: %s", clusterTemplate.Name)
+
+	tools.PrintResource(t, clusterTemplate)
+	tools.PrintResource(t, clusterTemplate.CreatedAt)
+
+	th.AssertEquals(t, name, clusterTemplate.Name)
+	th.AssertEquals(t, choices.ExternalNetworkID, clusterTemplate.ExternalNetworkID)
+	th.AssertEquals(t, choices.MagnumImageID, clusterTemplate.ImageID)
+
+	return clusterTemplate, nil
+}
+
+func createKubernetesCluster(t *testing.T, client *gophercloud.ServiceClient, clusterTemplateID string) (string, error) {
```

Comment:
Same here

Reply:
In that case rather than just duplicating the code and changing a few lines for the timeout I'd prefer to make it work properly.
I will make a separate PR for [these changes](https://github.com/tghartland/gophercloud/compare/master...tghartland:containerinfra-acceptance-timeout) which I can then use here.

I can't test it yet because a power cut broke my devstack, but I'll open that pull request as soon as possible.

Reply:
I've been able to test the changes I want to make in devstack on a virtual machine, it took a lot longer than the physical machine I was using so there will definitely need to be some tweaking to get the timeouts right for the openlab test environment.

The tests took 40 minutes to complete on a virtual machine devstack, if it's anything like that in the CI then it might be worth changing the existing cluster tests so that the node group tests are done half way through. This prevents having to create a second cluster just for the node group tests, but would sacrifice having clear separation of tests for the two separate APIs. 

To get the cluster tests working on Magnum master branch I also had to change the cluster template, for the same `docker_storage_driver` issue. Hopefully that won't cause problems with old magnum versions.

I have my physical machine devstack back now, the clusters CRUD test runs in 4 minutes thankfully.

I've opened a PR for the changes here: https://github.com/gophercloud/gophercloud/pull/1794.

Reply:
>  if it's anything like that in the CI then it might be worth changing the existing cluster tests so that the node group tests are done half way through.

My goal is to have the cluster tests run as a separate job, like the ironic tests are currently. That should help keep all testing to around 60-70 minutes.

> Hopefully that won't cause problems with old magnum versions.

If it does, I'll try to manage that. The current Magnum support has all been done ad-hoc against any cloud we could find, and then aggregating the results, so it's always been a little hacky.

Thanks so much for working on this.

Reply:
Done using the changes in #1794 to change the timeout for creating a kubernetes cluster. The timeouts will probably need to be adjusted once we get the results from the CI.